correct_programs/aoc2020/day_20_jurassic_jigsaw.py

```diff
@@ -54,6 +63,23 @@ def flip_horizontal(self) -> "Tile":
             reverse_side(self.right),
         )
 
+    def __repr__(self) -> str:
+        return (
+            f"top={self.top}, "
+            f"right={self.right}, "
+            f"bottom={self.bottom}, "
+            f"left={self.left}"
+        )
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, Tile):
+            return hash(self) == hash(other)
```

Comment:
This implementation of eq would be vulnerable to hash collision problems, yeah?

Reply:
Yes, that's correct. I'll fix it.

Reply:
Fixed in https://github.com/mristin/python-by-contract-corpus/pull/42